src/community/oseo/oseo-stac/src/test/java/org/geoserver/ogcapi/stac/ItemsTest.java

```diff
@@ -491,4 +496,97 @@ public void testQueryByDynamicProperty() throws Exception {
 
         assertThat((List<String>) doc.read("features[*].id"), contains("SAS1_20180226102021.01"));
     }
+
+    @Test
+    public void testPropertySelectionOnDynamicMerge() throws Exception {
+        DocumentContext result =
+                getAsJSONPath(
+                        "ogc/stac/collections/LANDSAT8/items?fields=properties,-assets,assets.thumbnail.type",
+                        200);
+
+        // empty fields param only mandatory attributes should be there.
+        String id = result.read("features[0].id");
+        String type = result.read("features[0].type");
+        Map<String, Object> geom = result.read("features[0].geometry");
+        JSONArray bbox = result.read("features[0].bbox");
+        Map<String, Object> properties = result.read("features[0].properties");
+        Map<String, Object> assets = result.read("features[0].assets");
+        String thumbnailType = result.read("features[0].assets.thumbnail.type");
```

Comment:
I believe the test should also check all other properties inside assets are gone, both in ``asserts.*`` and under ``asserts.thumbnail.*``

Reply:
fixed